deps/gen_functions.c

```diff
@@ -57,6 +58,7 @@ int main(int argc, char *argv[])
   printf("const MPI_WAITALL            = Libdl.dlsym(libmpi, \"%s\")\n", STRING(MPI_WAITALL));
   printf("const MPI_WAITANY            = Libdl.dlsym(libmpi, \"%s\")\n", STRING(MPI_WAITANY));
   printf("const MPI_WTIME              = Libdl.dlsym(libmpi, \"%s\")\n", STRING(MPI_WTIME));
+  printf("const MPI_IN_PLACE           = Base.box(Ptr{Int}, Base.unbox(Int, Int(%d)))\n", MPI_IN_PLACE);
```

Comment:
Julia uses the Fortran MPI interface. We need the Fortran values for all constants. We're calling the Fortran interface via `ccall`, but that's only an implementation detail.